### PR TITLE
Merge HealthKit to develop

### PR DESCRIPTION
**Pull request by markyou**

### CODE REVIEW

**Comment by gniezen** [comment]
@markyou I'm now realizing that it's not really feasible to try and do a code review of the entire PR (with 644 files changed). Maybe I should rather just focus on sections actually creating records and pushing them to the Tidepool API, as it's mostly data integrity we care about in this case. It looks like that would be `HealthKitDataUploader.swift`?

**Comment by markyou** [comment]
@gniezen Yes, the huge number of files includes pods that are installed with source checked in, which is a best-practice (unlike npm). Ignore all the pods stuff. Focus on:

HealthKitData.swift
HealthKitDataCache.swift
HealthKitDataManager.swift
HealthKitDataUploader.swift

APIConnect.swift
AppDelegate.swift
LoginViewController.swift

But, yes, primarily just HealthKitDataUploader.swift for review for data integrity concerns.

File: Urchin/HealthKit/HealthKitDataUploader.swift [hunk 1]
```diff
@@ -252,13 +252,40 @@ class HealthKitDataUploader {
...
+                    annotationValue = "high"
+                    annotationThreshold = 400
+                }
+                if let annotationCode = annotationCode,
+                       annotationValue = annotationValue {
+                    let annotations = [
+                        [
+                            "annotationCode": annotationCode,
+                            "annotationValue": annotationValue,
+                            "annotationThreshold": annotationThreshold
```
Comment (gniezen) [inline Urchin/HealthKit/HealthKitDataUploader.swift]: Should be `{"code": annotationCode, "value": annotationValue, "threshold": annotationThreshold}`?

Comment (markyou) [inline Urchin/HealthKit/HealthKitDataUploader.swift]: Yes, it should. Apologies. Was running on fumes after all nighter. Updating now. Apparently jellyfish doesn't validate the annotations schema? Samples with annotations uploaded without issue using my incorrect key names.

**Pull request merged**